docs/source/python/api/tables.rst

```diff
@@ -46,6 +46,18 @@ Classes
    TableGroupBy
    RecordBatchReader
 
+Dataframe Interchange Protocol
+------------------------------
+
+.. currentmodule:: pyarrow.interchange
+
+.. autosummary::
+   :toctree: ../generated/
+
+   from_dataframe
```

Comment:
Potential alternative might be to do `interchange.from_dataframe` here, to avoid the currentmodule change back and forth